modules/apps/portal-workflow/portal-workflow-kaleo-service/src/main/java/com/liferay/portal/workflow/kaleo/internal/model/listener/ObjectDefinitionModelListener.java

```diff
@@ -0,0 +1,181 @@
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Pedro Leite
+ */
+@Component(service = ModelListener.class)
+public class ObjectDefinitionModelListener
+	extends BaseModelListener<ObjectDefinition> {
+
+	@Override
+	public void onAfterUpdate(
+			ObjectDefinition originalModel, ObjectDefinition model)
+		throws ModelListenerException {
+
+		if (originalModel.getTitleObjectFieldId() !=
+				model.getTitleObjectFieldId()) {
+
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.fetchObjectDefinition(
+					model.getObjectDefinitionId());
+
+			for (ObjectEntry objectEntry :
+					_objectEntryService.getObjectEntries(
+						objectDefinition.getObjectDefinitionId())) {
+
+				KaleoInstance kaleoInstance =
+					KaleoInstanceUtil.fetchByCN_CPK_First(
+						objectDefinition.getClassName(),
+						objectEntry.getObjectEntryId(), null);
+
+				if (kaleoInstance != null) {
+					try {
+						String objectEntryTitleValue =
+							objectEntry.getTitleValue();
+
+						if (Validator.isNotNull(objectEntryTitleValue)) {
+							_instanceWorkflowMetricsIndexer.updateInstance(
+								kaleoInstance.isActive(),
+								_createAssetTitleLocalizationMap(
+									kaleoInstance.getClassName(),
+									kaleoInstance.getClassPK(),
+									kaleoInstance.getGroupId(),
+									objectEntryTitleValue),
+								objectDefinition.getLabelMap(),
+								kaleoInstance.getCompanyId(),
+								kaleoInstance.getKaleoInstanceId(),
+								kaleoInstance.getModifiedDate());
+						}
+					}
+					catch (PortalException portalException) {
+						throw new ModelListenerException(
+							portalException.getMessage());
+					}
+				}
+			}
+		}
+	}
+
+	private Map<Locale, String> _createAssetTitleLocalizationMap(
+		String className, long classPK, long groupId,
+		String objectEntryTitleValue) {
+
+		AssetRenderer<?> assetRenderer = _getAssetRenderer(className, classPK);
+
+		if (assetRenderer != null) {
+			AssetEntry assetEntry = _assetEntryLocalService.fetchEntry(
+				assetRenderer.getClassName(), assetRenderer.getClassPK());
+
+			if (assetEntry != null) {
+				return _localization.populateLocalizationMap(
+					HashMapBuilder.put(
+						LocaleUtil.fromLanguageId(
+							assetEntry.getDefaultLanguageId()),
+						objectEntryTitleValue
+					).build(),
+					assetEntry.getDefaultLanguageId(), assetEntry.getGroupId());
+			}
+		}
+
+		WorkflowHandler<?> workflowHandler =
+			WorkflowHandlerRegistryUtil.getWorkflowHandler(className);
+
+		if (workflowHandler != null) {
+			Map<Locale, String> localizationMap = new HashMap<>();
+
+			for (Locale availableLocale :
+					_language.getAvailableLocales(groupId)) {
+
+				localizationMap.put(
+					availableLocale,
+					workflowHandler.getTitle(classPK, availableLocale));
+			}
+
+			return localizationMap;
+		}
+
+		return Collections.emptyMap();
+	}
+
+	private AssetRenderer<?> _getAssetRenderer(String className, long classPK) {
```

Comment:
Since this method is used only once, you should bring this code to  **_createAssetTitleLocalizationMap(...)**